SonarQube.Client.Tests/Models/ServerSentEvents/SSEStreamReaderTests.cs

```diff
@@ -137,6 +137,47 @@ public async Task GetNextEventOrNullAsync_IssueChangedEventType_DeserializedEven
                     issues: new[] { new BranchAndIssueKey("key1", "master") }));
         }
 
+        [TestMethod]
+        public async Task GetNextEventOrNullAsync_TaintVulnerabilityClosedEventType_DeserializedEvent()
+        {
+            const string serializedTaintVulnerabilityClosedEvent =
+                "{\"projectKey\": \"projectKey1\",\"key\": \"taintKey\"}";
+
+            var channel = CreateChannelWithEvents(new SqServerEvent("TaintVulnerabilityClosed", serializedTaintVulnerabilityClosedEvent));
+
+            var testSubject = CreateTestSubject(sqEventsChannel: channel);
+
+            var result = await testSubject.GetNextEventOrNullAsync();
+
+            result.Should().NotBeNull();
+            result.Should().BeOfType<TaintVulnerabilityClosedServerEvent>();
+            result.Should().BeEquivalentTo(
+                new TaintVulnerabilityClosedServerEvent(
+                    projectKey: "projectKey1",
+                    key: "taintKey"));
+        }
+
+        [TestMethod]
+        public async Task GetNextEventOrNullAsync_TaintVulnerabilityRaisedEventType_DeserializedEvent()
+        {
+            const string serializedTaintVulnerabilityRaisedEvent =
```

Comment:
We actually get the entire payload back, and seems like it comes with the secondary location hashes too: https://github.com/SonarSource/sonarlint-core/blob/171ca4d75c24033e115a81bd7481427cd1f39f4c/server-api/src/test/java/org/sonarsource/sonarlint/core/serverapi/push/PushApiTests.java#L354

So we might not need to pull anything on SLVS side and could instead just deserialize the data from the event. I asked Damien about it but he's off today. For now I deserialized the other fields, will revisit it next week.